presto-main/src/main/java/com/facebook/presto/operator/ScanFilterAndProjectOperator.java

```diff
@@ -253,7 +254,7 @@ private Page processColumnSource()
 
         // only return a page if buffer is full or we are finishing
         Page page = null;
-        if (!pageBuilder.isEmpty() && (finishing || pageBuilder.isFull())) {
+        if (!pageBuilder.isEmpty() && (finishing || signal.isAborted() || pageBuilder.isFull())) {
```

Comment:
if `signal.isAborted()` we shouldn't bother with producing a page. 

OTOH, I understand why you'd like to check `isAborted()` in loops, but here's no loop and you're yielding anyway.